lib/hanami/model/configuration.rb

```diff
@@ -98,6 +99,14 @@ def define_entities_mappings(container, repositories)
           entity.schema = Sql::Entity::Schema.new(entities, container.relations[relation], mappings.fetch(relation))
         end
       end
+
+      # @since x.x.x
+      # @api private
+      def run_before_callbacks
+        @callbacks[:before].each do |cb|
```

Comment:
My only reservation is that `before` callbacks are limited to configuring the connection.  It may be more flexible to yield an object which has the connection as an accessor, but if you don't anticipate needing that functionality, then this is fine.